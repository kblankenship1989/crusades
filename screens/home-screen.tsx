import React from 'react';
import {View, Button, Modal, TextInput, Text, ColorSchemeName, FlatList} from 'react-native';

type HomeState = {
    isModalOpen: boolean
}

type HomeProps = {
    colorScheme: ColorSchemeName
}

export class HomeScreen extends React.Component<HomeProps, HomeState> {
    constructor(props : HomeProps) {
        super(props);

        this.state = {
            isModalOpen: false
        };
    }

    toggleModal = () : void => {
        this.setState({
            isModalOpen: !this.state.isModalOpen
        });
    }

    render() : JSX.Element {
        return (
            <View>
                <Button
                    title={'add-order-of-battle'}
                    onPress={() => null}
                    testID={'add-order-of-battle-button'}
                >
                    <Text>{'Add'}</Text>
                </Button>
                <FlatList
                    renderItem={null}
                    keyExtractor={(item, index) => index.toString()}
                    data={null}
                />
                <Modal
                    visible={this.state.isModalOpen}
                >
                    <View>
                        <TextInput
                            placeholder={'Title'}
                            onChangeText={() => null}
                            value={''}
                        />
                        <Button
                            title={'create-order-of-battle'}
                            onPress={() => null}
                            testID={'create-order-of-battle-button'}
                        >
                            <Text>{'Create'}</Text>
                        </Button>
                    </View>
                </Modal>
            </View>
        );
    }
}
