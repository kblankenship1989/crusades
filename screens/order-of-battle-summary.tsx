import React from 'react';
import {RouteProp} from '@react-navigation/native';
import {RootParamList} from '../types/root-param-list';
import {View} from 'react-native';
import {Input, Icon} from 'react-native-elements';
import {RequisitionPointsIcon} from '../components/requisition-points-icon';

type OrderOfBattleSummaryProps = {
    route: RouteProp<RootParamList, 'OrderOfBattleSummary'>
};

export const OrderOfBattleSummary = ({route} : OrderOfBattleSummaryProps) : JSX.Element => {
    const {
        title,
        faction,
        requisitionPoints
    } = route.params.orderOfBattle;

    return (
        <View>
            <Input
                leftIcon={
                    <Icon
                        size={18}
                        name={'format-title'}
                        type={'material-community'}
                    />
                }
                placeholder={'Title'}
                onChangeText={() => null}
                value={title}
                style={{
                    fontSize: 18,
                    margin: 10
                }}
            />
            <Input
                leftIcon={
                    <Icon
                        size={18}
                        name={'sword-cross'}
                        type={'material-community'}
                    />
                }
                placeholder={'Faction'}
                onChangeText={() => null}
                value={faction}
                style={{
                    fontSize: 18,
                    margin: 10
                }}
            />
            <RequisitionPointsIcon
                currentPoints={requisitionPoints}
            />
        </View>
    );
};
