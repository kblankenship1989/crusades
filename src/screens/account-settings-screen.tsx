import React from 'react';
import {ActionFixedFooterContainer} from '../containers/action-fixed-footer-container';

export type AccountSettingsProps = {
    selectedPlayer: Player,
    updatePlayer: (updates: Player) => void
}

type AccountSettingsState = Player & {
    isDirty: boolean
}

export class AccountSettings extends React.Component<AccountSettingsProps, AccountSettingsState> {
    render() : React.ReactNode {
        return (
            <ActionFixedFooterContainer>

            </ActionFixedFooterContainer>
        );
    }
}
