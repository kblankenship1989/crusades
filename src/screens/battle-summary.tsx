import React from 'react';
import {Button, FlatList, ListRenderItem, View} from 'react-native';
import {Divider, Text} from 'react-native-elements';
import {StackNavigationProp,} from '@react-navigation/stack';

import {ActionFixedFooterContainer} from '../containers/action-fixed-footer-container';
import {BackgroundImageListItem} from '../components/background-image-list-item';
import {defaultBattleResults} from '../redux/state/order-of-battle/battle-results';
import {BattleOutcomes, Factions} from '../enums';
import {FactionPicker} from '../components/faction-picker';
import {BattleOutcomePicker} from '../components/battle-outcome-picker';
import {RouteProp} from '@react-navigation/native';

export type BattleSummaryProps = {
    navigation: StackNavigationProp<RootParamList, 'BattleSummary'>
    route: RouteProp<RootParamList, 'BattleSummary'>
}

type BattleSummaryState = BattleResults & {
    currentResults: BattleResults[],
    isDirty: boolean,
    isNewEntryDirty: boolean
}

export class BattleSummary extends React.Component<BattleSummaryProps, BattleSummaryState> {
    constructor(props : BattleSummaryProps) {
        super(props);

        this.state = {
            ...defaultBattleResults,
            currentResults: props.route.params.battleResults,
            isDirty: false,
            isNewEntryDirty: false
        };
    }

    onSave = () : void => {
        this.props.navigation.navigate('OrderOfBattleSummary', {
            updatedBattleResults: this.state.currentResults
        });
    }

    selectEnemyFaction = (enemyFaction: Factions) : void => {
        this.setState({
            enemyFaction,
            isNewEntryDirty: true
        });
    }

    selectResult = (result: BattleOutcomes) : void => {
        this.setState({
            result,
            isNewEntryDirty: true
        });
    }

    removeBattleResult = (index: number) : void => {
        const newCurrentResults = [...this.state?.currentResults];
        newCurrentResults.splice(index,1);
        this.setState({
            currentResults: newCurrentResults,
            ...defaultBattleResults,
            isDirty: true
        });
    }

    addCurrentResult = () : void => {
        const newBattleResult : BattleResults = {
            enemyFaction: this.state.enemyFaction,
            date: new Date().toLocaleDateString(),
            result: this.state.result,
            markedForGreatness: this.state.markedForGreatness
        };

        const newCurrentResults = [...this.state?.currentResults];
        newCurrentResults.unshift(newBattleResult);

        this.setState({
            currentResults: newCurrentResults,
            ...defaultBattleResults,
            isDirty: true,
            isNewEntryDirty: false
        });
    }

    renderListItem : ListRenderItem<BattleResults> = ({item, index}) => {
        const title = `${item.enemyFaction} - ${item.result} (${item.date})`;

        return (
            <BackgroundImageListItem
                index={index}
                title={title}
                imageKey={item.enemyFaction}
                onDelete={this.removeBattleResult}
            />
        );
    };

    render() : React.ReactNode {
        return (
            <ActionFixedFooterContainer
                isDirty={this.state.isDirty}
                onSave={this.onSave}
            >
                <View>
                    <FactionPicker
                        selectedFaction={this.state.enemyFaction}
                        onChange={this.selectEnemyFaction}
                        title={'Enemy Faction'}
                    />
                    <BattleOutcomePicker
                        selectedBattleOutcome={this.state.result}
                        onChange={this.selectResult}
                    />
                    <Text h4>{'Mark For Greatness'}</Text>
                    <Button
                        title={'+ ADD'}
                        onPress={this.addCurrentResult}
                    />
                </View>
                <Divider/>
                <FlatList
                    ListHeaderComponent={<Text h3>{'Previous Battles'}</Text>}
                    renderItem={this.renderListItem}
                    keyExtractor={(battleResults, index) => index.toString()}
                    data={this.state.currentResults}
                />
            </ActionFixedFooterContainer>
        );
    }
}
