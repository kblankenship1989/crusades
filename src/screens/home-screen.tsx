import React from 'react';
import {FlatList, ListRenderItem} from 'react-native';
import {Text} from 'react-native-elements';
import {ScrollView} from 'react-native-gesture-handler';
import {ConnectedProps} from 'react-redux';
import {StackNavigationProp} from '@react-navigation/stack';

import {homeScreenConnector} from './home-screen-connector';
import {ActionFixedFooterContainer} from '../containers/action-fixed-footer-container';
import {BackgroundImageListItem} from '../components/background-image-list-item';

export type HomeProps = ConnectedProps<typeof homeScreenConnector> & {
    navigation: StackNavigationProp<RootParamList, 'Home'>
}

export const HomeScreen : React.FC<HomeProps> = ({
    loadSelectedOrderOfBattle,
    deleteSelectedOrderOfBattle,
    createOrderOfBattle,
    navigation,
    ordersOfBattle
}) => {
    const selectOrderOfBattle = (index: number) : void => {
        loadSelectedOrderOfBattle(index);
        navigation.push('OrderOfBattleSummary');
    };

    const addOrderOfBattle = () : void => {
        createOrderOfBattle();

        navigation.push('OrderOfBattleSummary');
    };

    const renderListItem : ListRenderItem<OrderOfBattle> = ({item, index}) => (
        <BackgroundImageListItem
            index={index}
            title={item.title || 'Untitled'}
            onPress={selectOrderOfBattle}
            onDelete={deleteSelectedOrderOfBattle}
            imageKey={item.faction}
        />
    );

    return (
        <ActionFixedFooterContainer
            onAdd={addOrderOfBattle}
        >
            {ordersOfBattle.length ?
                <FlatList
                    renderItem={renderListItem}
                    keyExtractor={(orderOfBattle) => orderOfBattle.id.toString()}
                    data={ordersOfBattle}
                />
                :
                <ScrollView>
                    <Text>{'Click + ADD to create a new Crusade Force'}</Text>
                </ScrollView>
            }
        </ActionFixedFooterContainer>
    );
};
