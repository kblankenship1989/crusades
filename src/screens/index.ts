export {AccountSummaryConnector} from './accounts/account-summary-connector';
export {EditPlayerConnector} from './accounts/edit-player-connector';
export {CrusadeCardListConnector} from './crusade-cards/crusade-card-list-connector';
export {CrusadeCardSummaryConnector} from './crusade-cards/crusade-card-summary-connector';
export {EditCrusadeCardConnector} from './crusade-cards/edit-crusade-card-connector';
export {EditOrderOfBattleConnector} from './orders-of-battle/edit-order-of-battle-connector';
export {OrderOfBattleSummaryConnector} from './orders-of-battle/order-of-battle-summary-connector';
export {OrdersOfBattleListConnector} from './orders-of-battle/orders-of-battle-list-connector';
