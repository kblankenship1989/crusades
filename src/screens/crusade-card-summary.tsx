import React from 'react';
import {View} from 'react-native';
import {CrusadeCardSummaryProps} from '../types/screens/props';
import {CrusadeCardSummaryState} from '../types/screens/states';


export class CrusadeCardSummary extends React.Component<CrusadeCardSummaryProps, CrusadeCardSummaryState> {
    constructor(props : CrusadeCardSummaryProps) {
        super(props);

        this.state = {
            ...props.currentCrusadeCard,
            isDirty: false,
            isEditing: false
        };
    }

    render() : JSX.Element {
        return <View/>;
    }
}
