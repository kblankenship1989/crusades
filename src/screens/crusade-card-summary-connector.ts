import {connect} from 'react-redux';
import {State} from '../redux/types/state';
import {saveCurrentCrusadeCard} from '../redux/action-creators/current-crusade-card';
import {CrusadeCardSummary} from './crusade-card-summary';

const mapStateToProps = (state: State) => ({
    currentCrusadeCard: state.currentCrusadeCard,
});

const mapDispatchToProps = {
    saveCurrentCrusadeCard
};

export const crusadeCardSummaryConnector = connect(mapStateToProps, mapDispatchToProps);
export const CrusadeCardSummaryConnector = crusadeCardSummaryConnector(CrusadeCardSummary);
