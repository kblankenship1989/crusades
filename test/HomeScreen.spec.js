import React from 'react';
import {fireEvent, render, waitFor, waitForElementToBeRemoved} from '@testing-library/react-native';
import {HomeScreen} from '../screens/home-screen';

describe('Creating an Order of Battle', () => {
    let component;
    
    beforeEach(() => {
        component = render(<HomeScreen />);
    });

    it('should add the Order of Battle to the list', () => {
        const title = 'Test Order of Battle';

        const addButton = component.getByTestId('add-order-of-battle-button');
        fireEvent.press(addButton);

        const oOBTitle = component.getByPlaceholderText('Title');
        fireEvent.changeText(oOBTitle, title);

        const createButton = component.getByTestId('create-order-of-battle-button');
        fireEvent.press(createButton);

        fireEvent.press(addButton);

        fireEvent.changeText(oOBTitle, title);

        fireEvent.press(createButton);

        expect(component.queryAllByTestId('order-of-battle')).toHaveLength(2);

        expect(component.toJSON()).toMatchSnapshot();
    });
});
