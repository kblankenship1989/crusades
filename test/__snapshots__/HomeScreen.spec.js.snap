// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Creating an Order of Battle should add the Order of Battle to the list 1`] = `
<View>
  <View
    accessibilityRole="button"
    accessibilityState={Object {}}
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
    testID="add-order-of-battle-button"
  >
    <View
      style={
        Array [
          Object {},
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#007AFF",
              "fontSize": 18,
              "margin": 8,
              "textAlign": "center",
            },
          ]
        }
      >
        add-order-of-battle
      </Text>
    </View>
  </View>
  <Modal
    hardwareAccelerated={false}
    visible={false}
  >
    <View>
      <TextInput
        allowFontScaling={true}
        onChangeText={[Function]}
        placeholder="Title"
        rejectResponderTermination={true}
        underlineColorAndroid="transparent"
        value=""
      />
      <View
        accessibilityRole="button"
        accessibilityState={Object {}}
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
        testID="create-order-of-battle-button"
      >
        <View
          style={
            Array [
              Object {},
            ]
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#007AFF",
                  "fontSize": 18,
                  "margin": 8,
                  "textAlign": "center",
                },
              ]
            }
          >
            create-order-of-battle
          </Text>
        </View>
      </View>
    </View>
  </Modal>
</View>
`;

exports[`Creating an Order of Battle should add the Order of Battle to the list 2`] = `
<View>
  <View
    accessibilityRole="button"
    accessibilityState={Object {}}
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
    testID="add-order-of-battle-button"
  >
    <View
      style={
        Array [
          Object {},
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#007AFF",
              "fontSize": 18,
              "margin": 8,
              "textAlign": "center",
            },
          ]
        }
      >
        add-order-of-battle
      </Text>
    </View>
  </View>
  <Modal
    hardwareAccelerated={false}
    visible={false}
  >
    <View>
      <TextInput
        allowFontScaling={true}
        onChangeText={[Function]}
        placeholder="Title"
        rejectResponderTermination={true}
        underlineColorAndroid="transparent"
        value=""
      />
      <View
        accessibilityRole="button"
        accessibilityState={Object {}}
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
        testID="create-order-of-battle-button"
      >
        <View
          style={
            Array [
              Object {},
            ]
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#007AFF",
                  "fontSize": 18,
                  "margin": 8,
                  "textAlign": "center",
                },
              ]
            }
          >
            create-order-of-battle
          </Text>
        </View>
      </View>
    </View>
  </Modal>
</View>
`;
